### DISCUSSION TITLE
from gnome 41 to gnome 42 - strange behavior and strange workflow now

### DISCUSSION THREAD

**Discussion by alansenairj** [flameshot-org/flameshot#2690] [Q&A] [answered]
Now I need to click on Flameshot and then it captures a screen and I need to "Share this capture with an app" to open editor. 

What a mess! On later Gnone 41 I just click on it and I have two thing: I send me to selection and go to clipboard automatically and I did not have any sounds on it. 

Someone knows how to setup new falmeshot 11 to goes directly to editor and clipboard and remove sound of  screen captures?

**Comment by mmahmoudian** [comment, marked as answer]
Please try [searching in the issues](https://github.com/flameshot-org/flameshot/issues?q=is%3Aissue+gnome+share) or at least check the pinned issues (image below) before creating a new issue or discussion. This way you at least get to the answer faster. This post is the 54th duplicate on the same topic, meaning you could practically use any search terms and get to the right answer (link above, image below).

Please take your **rightful complaints** to Gnome developers. This is what they decided to do. For more information, read this link on our website:

https://flameshot.org/docs/guide/wayland-help/#i-am-asked-to-share-my-screen-every-time

The "mess" you are talking about is the direct result of having Gnome Wayland. Switch your DE to literally *anything else* (KDE, Mate, Solus, LXQt, Xfce, ....) and you will not experience this.

![image](https://user-images.githubusercontent.com/390889/174633885-0cb5a983-607a-4b12-9d5e-d33ea5ef0b87.png)

**Comment by alansenairj** [comment]
Man! I repeat! What a mess! But is not your fault, sorry. 
But you cannot assume I knew the correct words cause it is "gnome 42" + "fedora 36" + "I cannot describe this behavior" issue. Next time someone else can find this words too. Take a break! 

Thanx to inform me about this stupid security implement done by Gnome guys. I loose time trying to solve it and I did a roll back  to my OS back-up "Fedora 35 + Gnome 41" yesterday.
Maybe when Gnome be in 44 and I will do an upgrade, maybe. 

Thanxx for your patience.

**Comment by alansenairj** [comment]
Bro I Think you guys must coordinate yourself 

[Florian Müllner](https://gitlab.gnome.org/fmuellner) [commented](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5606#note_1484610):

I think this may be related to the gnome-screenshot breakage we are seeing in https://bugzilla.redhat.com/show_bug.cgi?id=2009460:

This was fixed in [gnome-screenshot#179 (closed)](https://gitlab.gnome.org/GNOME/gnome-screenshot/-/issues/179).

I suspect this also broke flameshot on gnome41

That is what you may get if you rely on internal APIs. The xdg portal does provide a public screenshot API to applications with versioning and a stability promise.

But as a software engineer I sure heard about API/endpoints versioning and backward compatibility concerns

That applies to public API. Here we are talking about an API that was only ever intended to be between the compositor and core components like gnome-settings-daemon/gnome-control-center. If you want an official API with versioning and backwards compatibility, then there is a public API in xdg-desktop-portal. If you insist on using non-public API, you don't get to complain if you end up having to pick up the pieces.

**Comment by alansenairj** [comment]
What is that "strange permitions" ? 

He is talking about a public API! 
[Florian Müllner](https://gitlab.gnome.org/fmuellner) [commented](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5605#note_1484622):

I cannot use capture software without ask me a strange permission

The problem is not the "strange permission", but that the software in question tries to use private API (between gnome-shell and gnome-settings-daemon/gnome-control-center/xdg-desktop-portal-gnome).

For applications, xdg-desktop-portal provides a public API that doesn't require "strange permissions".
